precompile/erc721/src/lib.rs

```diff
@@ -42,22 +47,30 @@ impl<AssetManager> Erc721Precompile<AssetManager>
 where
 	AssetManager: pallet_living_assets_ownership::traits::Erc721,
 {
-	fn owner_of(handle: &mut impl PrecompileHandle) -> EvmResult<PrecompileOutput> {
+	fn owner_of(
+		collection_id: CollectionId,
+		handle: &mut impl PrecompileHandle,
+	) -> EvmResult<PrecompileOutput> {
 		let mut input = handle.read_input()?;
 		input.expect_arguments(1)?;
 
 		let asset_id: U256 = input.read()?;
 
-		// collection id is encoded into the contract address
-		let collection_id = match address_to_collection_id(handle.code_address()) {
```

Comment:
given that we don't assume the `collection_id` is the same as the contract we are instantiating anymore, could we remove `address_to_collection_id` func at all?